src/query-performance-monitoring/common-parameters/common_parameters.go

```diff
@@ -0,0 +1,53 @@
+package commonparameters
+
+import (
+	"github.com/newrelic/infra-integrations-sdk/v3/log"
+	"github.com/newrelic/nri-postgresql/src/args"
+)
+
+// The maximum number records that can be fetched in a single metrics
+const MaxQueryCountThreshold = 30
+
+// DefaultQueryMonitoringCountThreshold is the default threshold for the number of queries to monitor.
+const DefaultQueryMonitoringCountThreshold = 20
+
+// DefaultQueryResponseTimeThreshold is the default threshold for the response time of a query.
+const DefaultQueryResponseTimeThreshold = 500
+
+type CommonParameters struct {
+	Version                              uint64
+	Databases                            string
+	Arguments                            args.ArgumentList
+	QueryMonitoringCountThreshold        int
+	QueryMonitoringResponseTimeThreshold int
+}
+
+func SetCommonParameters(args args.ArgumentList, version uint64, databases string) *CommonParameters {
```

Comment:
I would try to make SetCommonParameters go away. If I remember correctly it was enough to compute connectionInfo just one and pass that